### PR TITLE
bookened token

### PR DESCRIPTION
**Pull request by hadasz**
**Related Issue**  
Supports #95

**What does this PR do?**  
bookend token
**Description of Changes**  
bookend token

**What gif most accurately describes how I feel towards this PR?**  
Example of a gif

### CODE REVIEW

**Comment by hadasz** [comment]
Why was this moved to "in progress?" Doesn't a pull request waiting for code review belong in the review/QA column?

File: packages/bootleg-tokens/contracts/BookendSharedRoyaltyToken.sol [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    maxCount = maxCount > token.payments.length ? token.payments.length : maxCount;
+    uint256 payment = 0;
+    if (franchisor == _tokens[tokenId].franchisors[0]) {
+      for (uint256 i = start; i<maxCount; i++) {
+        payment = payment + (token.payments[i] * (100 - franchisorPercentage))/100;
+      }
+    } 
+    for (uint256 i = start; i < maxCount; i += 1) {
+      if (token.franchisors[i - 1] == franchisor) {
+        payment += (token.payments[i] * franchisorPercentage) % 100 == 0 ? (token.payments[i] * franchisorPercentage)/100  : (token.payments[i] * franchisorPercentage)/100 + 1;
```
Comment (breakpointer) [inline packages/bootleg-tokens/contracts/BookendSharedRoyaltyToken.sol]: This line is a little hard to read. Functionally correct but my coding style preference is to only use in-line ternary statements for simple logic.

File: packages/bootleg-tokens/test/BookendSharedRoyaltyToken.test.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+);
+
+contract('BookendSharedRoyaltyToken', accounts => {
+  const accountOne = accounts[0];
+  const accountTwo = accounts[1];
+  const accountThree = accounts[2];
+  const tokenId = 1;
+
+  //count is less than payment.length
+  it('count->minter->franchisor->integer', async () => {
```
Comment (breakpointer) [inline packages/bootleg-tokens/test/BookendSharedRoyaltyToken.test.js]: I'm not familiar with this test description notation so it's hard for me to understand what you're testing for in these blocks. Is this based on a standard?
My preference is to make the test descriptors almost like english sentences (e.g. `it('returns the default value when  count is less than payment array length', async () => {`)
While it is more verbose (and often redundant) it helps people not familiar with the code understand what the code should do. I find that often that person "not familiar with the code" is me after about a week 😄

**Comment by breakpointer** [comment]
> Why was this moved to "in progress?" Doesn't a pull request waiting for code review belong in the review/QA column?

Not sure. I don't recall when/why this happened.

File: packages/bootleg-tokens/test/BookendSharedRoyaltyToken.test.js (hunk 2 above)
Comment (hadasz) [reply packages/bootleg-tokens/test/BookendSharedRoyaltyToken.test.js]: Yea - this was linked to test coverage if, then statements. Was mainly for my own sake. I agree its not clear. Will change if there is time. Thanks for reviewing and merging!

File: packages/bootleg-tokens/contracts/BookendSharedRoyaltyToken.sol (hunk 1 above)
Comment (hadasz) [reply packages/bootleg-tokens/contracts/BookendSharedRoyaltyToken.sol]: Hm, line 38 is a ternary statement, won't nested ternary statements also be hard to read?

**Pull request merged**